teku/src/main/java/tech/pegasys/teku/cli/slashingprotection/RepairCommand.java

```diff
@@ -0,0 +1,160 @@
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.services.beaconchain.WeakSubjectivityInitializer;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.datastructures.state.AnchorPoint;
+
+@CommandLine.Command(
+    name = "repair",
+    description = "Repair slashing protection data files used by teku.",
+    mixinStandardHelpOptions = true,
+    abbreviateSynopsis = true,
+    versionProvider = PicoCliVersionProvider.class,
+    synopsisHeading = "%n",
+    descriptionHeading = "%nDescription:%n%n",
+    optionListHeading = "%nOptions:%n",
+    footerHeading = "%n",
+    footer = "Teku is licensed under the Apache License 2.0")
+public class RepairCommand implements Runnable {
+  public static final SubCommandLogger SUB_COMMAND_LOG = new SubCommandLogger();
+
+  @CommandLine.Mixin(name = "Data")
+  private ValidatorClientDataOptions dataOptions;
+
+  @CommandLine.Mixin(name = "Network")
+  private Eth2NetworkOptions eth2NetworkOptions;
+
+  @CommandLine.Option(
+      names = {"--check-only-enabled"},
+      paramLabel = "<BOOLEAN>",
+      description = "Read and report potential problems, but don't update anything.",
+      arity = "0..1")
+  private boolean checkOnlyEnabled = false;
+
+  @CommandLine.Option(
+      names = {"--update-all-enabled"},
+      paramLabel = "<BOOLEAN>",
+      description = "All validator slashing protection records should be updated.",
+      arity = "0..1")
+  private boolean updateAllEnabled = false;
+
+  @CommandLine.Option(
+      names = {"--with-slot"},
+      paramLabel = "<INTEGER>",
+      description =
+          "Update slashing protection files to contain the specified slot as a minimum."
+              + "Should be a future slot, or after when the validators stopped performing duties.\n"
+              + "NOTE: This can be calculated for most networks, and is generally not required.",
+      arity = "1")
+  private UInt64 suppliedSlot;
+
+  private final WeakSubjectivityInitializer wsInitializer = new WeakSubjectivityInitializer();
+
+  @Override
+  public void run() {
+    final Path slashProtectionPath =
+        SlashingProtectionCommandUtils.getSlashingProtectionPath(dataOptions);
+    SlashingProtectionCommandUtils.verifySlashingProtectionPathExists(
+        SUB_COMMAND_LOG, slashProtectionPath);
+    final Spec spec = eth2NetworkOptions.getNetworkConfiguration().getSpec();
+
+    final Optional<AnchorPoint> initialAnchor =
+        wsInitializer.loadInitialAnchorPoint(
+            spec, eth2NetworkOptions.getNetworkConfiguration().getInitialState());
+
+    final UInt64 computedSlot = getComputedSlot(initialAnchor, spec);
+    final UInt64 computedEpoch =
+        spec.atSlot(computedSlot).miscHelpers().computeEpochAtSlot(computedSlot);
+    final SlashingProtectionRepairer repairer =
+        new SlashingProtectionRepairer(SUB_COMMAND_LOG, updateAllEnabled);
+
+    repairer.initialise(slashProtectionPath);
```

Comment:
I suspect this should actually be combined into a single static `SlashingProtectionRepairer.create` method (`initialise` would still exist but just be a private method).  That way you can't create a partially inited object.